OADP-4001 Self-Service (#89043)
openshift/openshift-docs: modules/oadp-self-service-troubleshoot-nabsl-same-ns.adoc

// Module included in the following assemblies:
//
// backup_and_restore/application_backup_and_restore/oadp-self-service/oadp-self-service-troubleshooting.adoc

:_mod-docs-content-type: CONCEPT
[id="oadp-self-service-troubleshoot-nabsl-same-ns_{context}"]
= Error NonAdminBackupStorageLocation not found in the namespace

> Consider the following scenario:

Reviewer: ```suggestion
Consider the following scenario:
```
IMO this sentence reads incomplete. Can you add a small detail to it like -- consider this scenario in case of what or to do/see what?

Author: changed it to `Consider the following scenario of a namespace `admin` backup:`

> * You have created two `NonAdminBackupStorageLocations` (NABLs) in two different namespaces, for example, `nabsl-1` in `namespace-1` and `nabsl-2` in `namespace-2`.

Reviewer: ```suggestion
* You have created two `NonAdminBackupStorageLocations` (NABLs) custom resources (CRs) in two different namespaces, for example, `nabsl-1` in `namespace-1` and `nabsl-2` in `namespace-2`.
```

* You are taking a backup of `namespace-1` and use `nabsl-2` in the `NonAdminBackup` (NAB) CR.

> In this scenario, you get the error, `NonAdminBackupStorageLocation not found in the namespace: NonAdminBackupStorageLocation.oadp.openshift.io`, when you create the NAB CR. See example error.

Reviewer: ```suggestion
In this scenario, after creating the NAB CR, you get the following error:

[source,text]
----
NonAdminBackupStorageLocation not found in the namespace: NonAdminBackupStorageLocation.oadp.openshift.io
---- 
```

> You get the error because you are using a NABSL that belongs to a different namespace than the namespace that you are trying to back up.

Reviewer: ```suggestion
The cause of the error is that the NABSL CR does not belong to the namespace that you are trying to back up. 
```

Suggested the above change. Else is given in the .solution section

.Error

[source, yaml]
----
apiVersion: oadp.openshift.io/v1alpha1
kind: NonAdminBackup
...
status:
  conditions:
  - lastTransitionTime: "2025-02-20T10:13:00Z"
  message: 'NonAdminBackupStorageLocation not found in the namespace: NonAdminBackupStorageLocation.oadp.openshift.io
    "nabsl2" not found'
  reason: InvalidBackupSpec
  status: "False"
  type: Accepted
  phase: BackingOff
----

.Solution

Use the NABSL that belongs to the same namespace that you are trying to back up. 

> In the scenario explained in the error, you must use `nabsl-1` in the NAB CR, because you are backing up `namespace-1`.

Reviewer: ```suggestion
In this scenario, you must use `nabsl-1` in the NAB CR to back up `namespace-1`.
```
